### ISSUE TITLE
Dashboard fails to load database properties intermittently

### ISSUE THREAD

**Issue by cheenamalhotra** [microsoft/azuredatastudio#21766]
- Azure Data Studio Version: 1.41.1-rc1, Insiders
- OS Version: Windows 11

Below error occurs in STS randomly for Local instances as well as Azure SQL databases. In below screenshot, connection is made to `localhost`. Navigating out of Dashboard and re-opening it or reloading ADS resolves it and it's nature is random.

The stacktrace in the screenshot does not link to recently fixed `#dso` related queries, and may provide insights.

![image](https://user-images.githubusercontent.com/13396919/215232220-ec0fe804-dc99-4e0b-a0a6-76dd296a3633.png)

**Comment by cheenamalhotra** [comment]
Also to mention, this error also occurs when expanding OE nodes, leading to "Error: Failed to expand node" (#21691)

Stacktrace for OE:

```
23-01-31 14:48:36.1364825 pid:39092 tid:90 sqltools Error: 0 : Unexpected exception on the binding queue: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.SqlServer.Management.Smo.Database.get_IsSqlDw()
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.ServerVersionHelper.GetValidForFlag(SqlServerType serverType, Database database) in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\SqlServerType.cs:line 42
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel.DatabaseTreeNode.EnsureContextInitialized() in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\SmoModel\DatabaseTreeNode.cs:line 51
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel.SmoTreeNode.GetContext() in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\SmoModel\SmoTreeNode.cs:line 107
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Nodes.TreeNode.GetContextAs[T]() in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\Nodes\TreeNode.cs:line 313
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel.DatabaseTreeNode.PopulateChildren(Boolean refresh, String name, CancellationToken cancellationToken, String accessToken) in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\SmoModel\DatabaseTreeNode.cs:line 57
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Nodes.TreeNode.Refresh(CancellationToken cancellationToken, String accessToken) in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\Nodes\TreeNode.cs:line 270
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.ObjectExplorerService.<>c__DisplayClass35_0.<QueueExpandNodeRequest>b__1(IBindingContext bindingContext, CancellationToken cancelToken) in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\ObjectExplorerService.cs:line 447
   at Microsoft.SqlTools.ServiceLayer.LanguageServices.BindingQueue`1.<>c__DisplayClass34_1.<ProcessQueue>b__1() in D:\a\_work\1\s\src\Microsoft.SqlTools.ServiceLayer\LanguageServices\BindingQueue.cs:line 346
```

**Comment by alanrenmsft** [comment]
the error pop up issue has been fixed, will use this issue to track the actual error.

**Comment by cheenamalhotra** [comment]
Issue has been fixed with #21903 as I was able to identify and resolve sqltools errors that were related to Azure MFA authentication.
